# 06 Entrenamiento y Validación del Modelo (Google Colab)
*Cuaderno para ejecutar en Google Colab*


## Justificación del Entrenamiento en Google Colab

Debido a las **limitaciones de recursos de hardware disponibles localmente**, particularmente la **ausencia de una Unidad de Procesamiento Gráfico (GPU)**, el entrenamiento del modelo de detección de objetos YOLOv8n-obb se realizará en la plataforma **Google Colab**. Colab proporciona acceso gratuito a GPUs potentes, como la Tesla T4, que son **esenciales para acelerar significativamente el proceso de entrenamiento** de modelos de aprendizaje profundo. Entrenar un modelo complejo como YOLOv8 en una CPU local llevaría tiempos prohibitivamente largos, haciendo inviable la experimentación y el ajuste eficiente del modelo. Por lo tanto, el uso de la infraestructura de Google Colab **optimiza el tiempo de entrenamiento y permite aprovechar la potencia de las GPUs** para lograr resultados en un plazo razonable.

In [1]:
from google.colab import drive
drive.mount('/content/drive')

# Navegar a tu proyecto
%cd /content/drive/My\ Drive/TFM_BALSAS

Mounted at /content/drive
/content/drive/My Drive/TFM_BALSAS


## GPU

Nos aseguramos que tenemos acceso a GPU. Usamos `nvidia-smi` para hacerlo.

In [3]:
!nvidia-smi

Sat Apr  5 12:42:15 2025       
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 550.54.15              Driver Version: 550.54.15      CUDA Version: 12.4     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  Tesla T4                       Off |   00000000:00:04.0 Off |                    0 |
| N/A   35C    P8              9W /   70W |       0MiB /  15360MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+-----

In [4]:
import os
HOME = os.getcwd()
print(HOME)

/content/drive/MyDrive/TFM_BALSAS


## Importar librerias necesarias

In [5]:
# Pip install method (recommended)
!pip install ultralytics --upgrade -q

from IPython import display
display.clear_output()

import ultralytics
ultralytics.checks()

Ultralytics 8.3.102 🚀 Python-3.11.11 torch-2.6.0+cu124 CUDA:0 (Tesla T4, 15095MiB)
Setup complete ✅ (2 CPUs, 12.7 GB RAM, 40.9/112.6 GB disk)


In [6]:
from ultralytics import YOLO

from IPython.display import display, Image

## Entrenamiento personalizado

In [11]:
# Cargar el modelo preentrenado
model = YOLO('/content/drive/My Drive/TFM_BALSAS/yolov8n-obb.pt')

In [12]:
%%time
# Entrenar el modelo
results = model.train(
    data="dataset/data.yaml",    # Ruta al archivo YAML
    epochs=100,                  # Aumentado para mejor convergencia
    patience=15,                 # Mayor paciencia para evitar paradas prematuras
    imgsz=640,                   # Tamaño estándar para YOLOv8
    batch=8,                     # Reducido para evitar problemas de memoria en T4
    optimizer="AdamW",           # Optimizer más efectivo para detección de objetos
    lr0=0.01,                    # Learning rate inicial
    lrf=0.001,                   # Learning rate final
    momentum=0.937,              # Momentum para el optimizador
    weight_decay=0.0005,         # Weight decay para reducir overfitting
    warmup_epochs=3,             # Epochs de calentamiento
    warmup_momentum=0.8,         # Momentum durante el calentamiento
    warmup_bias_lr=0.1,          # Learning rate para bias durante calentamiento
    box=7.5,                     # Peso para pérdida de bounding box
    cls=0.5,                     # Peso para pérdida de clasificación
    dfl=1.5,                     # Peso para pérdida de distribución
    plots=True,                  # Generar gráficos
    save=True,                   # Guardar checkpoints
    cache=False,                 # No usar cache (para evitar problemas de memoria)
    device=0,                    # Usar GPU
    workers=4,                   # Aumentado ligeramente para mejor paralelismo
    project="deteccion_balsas",  # Nombre del proyecto
    name="yolov8n_obb_run1",     # Nombre del experimento
    exist_ok=True                # Continuar entrenamientos interrumpidos
)

Ultralytics 8.3.102 🚀 Python-3.11.11 torch-2.6.0+cu124 CUDA:0 (Tesla T4, 15095MiB)
engine/trainer: task=obb, mode=train, model=/content/drive/My Drive/TFM_BALSAS/yolov8n-obb.pt, data=dataset/data.yaml, epochs=100, time=None, patience=15, batch=8, imgsz=640, save=True, save_period=-1, cache=False, device=0, workers=4, project=deteccion_balsas, name=yolov8n_obb_run1, exist_ok=True, pretrained=True, optimizer=AdamW, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=Fals

100%|██████████| 755k/755k [00:00<00:00, 18.3MB/s]


Overriding model.yaml nc=15 with nc=1

                   from  n    params  module                                       arguments                     
  0                  -1  1       464  ultralytics.nn.modules.conv.Conv             [3, 16, 3, 2]                 
  1                  -1  1      4672  ultralytics.nn.modules.conv.Conv             [16, 32, 3, 2]                
  2                  -1  1      7360  ultralytics.nn.modules.block.C2f             [32, 32, 1, True]             
  3                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]                
  4                  -1  2     49664  ultralytics.nn.modules.block.C2f             [64, 64, 2, True]             
  5                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]               
  6                  -1  2    197632  ultralytics.nn.modules.block.C2f             [128, 128, 2, True]           
  7                  -1  1    295424  ultralytics

train: Scanning /content/drive/My Drive/TFM_BALSAS/dataset/labels/train... 581 images, 0 backgrounds, 0 corrupt: 100%|██████████| 581/581 [04:29<00:00,  2.15it/s]

train: WARNING ⚠️ /content/drive/My Drive/TFM_BALSAS/dataset/images/train/88.tif: 1 duplicate labels removed


train: New cache created: /content/drive/My Drive/TFM_BALSAS/dataset/labels/train.cache
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, num_output_channels=3, method='weighted_average'), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))


val: Scanning /content/drive/My Drive/TFM_BALSAS/dataset/labels/val... 166 images, 0 backgrounds, 0 corrupt: 100%|██████████| 166/166 [01:09<00:00,  2.38it/s]

val: WARNING ⚠️ /content/drive/My Drive/TFM_BALSAS/dataset/images/val/259.tif: 1 duplicate labels removed
val: WARNING ⚠️ /content/drive/My Drive/TFM_BALSAS/dataset/images/val/594.tif: 1 duplicate labels removed
val: New cache created: /content/drive/My Drive/TFM_BALSAS/dataset/labels/val.cache


Plotting labels to deteccion_balsas/yolov8n_obb_run1/labels.jpg... 
optimizer: AdamW(lr=0.01, momentum=0.937) with parameter groups 63 weight(decay=0.0), 73 weight(decay=0.0005), 72 bias(decay=0.0)
TensorBoard: model graph visualization added ✅
Image sizes 640 train, 640 val
Using 2 dataloader workers
Logging results to deteccion_balsas/yolov8n_obb_run1
Starting training for 100 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      1/100       1.2G      1.709        1.9      2.217          5        640: 100%|██████████| 73/73 [00:28<00:00,  2.58it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:06<00:00,  3.42it/s]

                   all        166        224      0.605      0.621      0.579       0.28



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      2/100      1.22G      1.857      1.668      2.487         13        640: 100%|██████████| 73/73 [00:21<00:00,  3.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:06<00:00,  3.09it/s]


                   all        166        224      0.605      0.575      0.564      0.266

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      3/100      1.34G      1.832      1.596      2.508          9        640: 100%|██████████| 73/73 [00:23<00:00,  3.13it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:06<00:00,  3.41it/s]

                   all        166        224      0.496      0.679      0.477      0.206



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      4/100      1.36G      1.932      1.687      2.591          7        640: 100%|██████████| 73/73 [00:21<00:00,  3.37it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:06<00:00,  3.26it/s]


                   all        166        224      0.699       0.71      0.676      0.323

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      5/100      1.38G      1.799      1.515      2.515         12        640: 100%|██████████| 73/73 [00:21<00:00,  3.40it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:06<00:00,  3.39it/s]

                   all        166        224      0.641      0.652      0.618       0.34



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      6/100      1.39G       1.74       1.41      2.404          8        640: 100%|██████████| 73/73 [00:21<00:00,  3.36it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:06<00:00,  3.40it/s]

                   all        166        224      0.611      0.433      0.493      0.267



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      7/100      1.41G      1.774      1.404      2.395          9        640: 100%|██████████| 73/73 [00:23<00:00,  3.10it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:05<00:00,  3.92it/s]

                   all        166        224      0.709      0.621      0.659      0.339



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      8/100      1.42G      1.746      1.454      2.374          6        640: 100%|██████████| 73/73 [00:22<00:00,  3.25it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:05<00:00,  3.91it/s]

                   all        166        224      0.801      0.688      0.733      0.388



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      9/100      1.45G       1.69      1.433      2.434         12        640: 100%|██████████| 73/73 [00:22<00:00,  3.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:05<00:00,  3.89it/s]

                   all        166        224      0.857      0.746       0.77      0.424



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     10/100      1.45G       1.68      1.369      2.383          6        640: 100%|██████████| 73/73 [00:22<00:00,  3.27it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.27it/s]

                   all        166        224      0.818      0.724      0.755      0.415



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     11/100      1.48G      1.731      1.394      2.475          6        640: 100%|██████████| 73/73 [00:22<00:00,  3.28it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.48it/s]

                   all        166        224       0.75      0.768      0.766      0.394



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     12/100       1.5G      1.663      1.343      2.408         13        640: 100%|██████████| 73/73 [00:22<00:00,  3.20it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.67it/s]

                   all        166        224      0.823      0.732      0.769      0.427



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     13/100      1.52G      1.657      1.325      2.411          9        640: 100%|██████████| 73/73 [00:23<00:00,  3.13it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.72it/s]

                   all        166        224      0.826      0.696      0.771      0.409



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     14/100      1.53G      1.637      1.315      2.387         16        640: 100%|██████████| 73/73 [00:23<00:00,  3.08it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.99it/s]

                   all        166        224      0.799       0.75      0.784       0.44



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     15/100      1.55G      1.694      1.307      2.405          6        640: 100%|██████████| 73/73 [00:23<00:00,  3.10it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.84it/s]

                   all        166        224      0.841      0.768      0.795      0.438



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     16/100      1.57G      1.633      1.265      2.306         11        640: 100%|██████████| 73/73 [00:23<00:00,  3.09it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.88it/s]

                   all        166        224      0.775      0.728      0.732      0.413



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     17/100      1.58G       1.64      1.294      2.342         14        640: 100%|██████████| 73/73 [00:23<00:00,  3.06it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.81it/s]

                   all        166        224      0.781      0.733      0.785      0.452



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     18/100      1.59G      1.608      1.254      2.299          4        640: 100%|██████████| 73/73 [00:23<00:00,  3.08it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.78it/s]

                   all        166        224      0.814      0.781      0.776      0.424



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     19/100      1.62G       1.63      1.283      2.372         12        640: 100%|██████████| 73/73 [00:23<00:00,  3.08it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.87it/s]

                   all        166        224      0.776      0.763      0.762       0.43



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     20/100      1.64G      1.607       1.33      2.357         10        640: 100%|██████████| 73/73 [00:23<00:00,  3.09it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.92it/s]

                   all        166        224      0.837      0.737      0.777      0.421



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     21/100      1.65G        1.6      1.259      2.367          8        640: 100%|██████████| 73/73 [00:25<00:00,  2.89it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.69it/s]

                   all        166        224      0.834      0.723      0.789       0.44



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     22/100      1.67G       1.57      1.243       2.32          9        640: 100%|██████████| 73/73 [00:23<00:00,  3.08it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  5.00it/s]

                   all        166        224      0.841       0.75      0.802      0.434



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     23/100      1.69G      1.524      1.193      2.275          8        640: 100%|██████████| 73/73 [00:23<00:00,  3.12it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.98it/s]

                   all        166        224      0.842      0.787      0.812      0.434



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     24/100      1.71G      1.609      1.296      2.351         10        640: 100%|██████████| 73/73 [00:23<00:00,  3.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.31it/s]

                   all        166        224      0.868       0.82      0.821      0.448



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     25/100      1.73G      1.605      1.233      2.317         13        640: 100%|██████████| 73/73 [00:22<00:00,  3.24it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.21it/s]

                   all        166        224      0.874      0.776      0.817      0.469



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     26/100      1.73G      1.572      1.173      2.312         10        640: 100%|██████████| 73/73 [00:22<00:00,  3.22it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:05<00:00,  3.99it/s]

                   all        166        224      0.832      0.797      0.807      0.455



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     27/100      1.76G      1.542      1.192       2.28          9        640: 100%|██████████| 73/73 [00:21<00:00,  3.34it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:05<00:00,  3.69it/s]

                   all        166        224      0.885      0.757      0.823      0.458



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     28/100      1.78G      1.532      1.171      2.276         20        640: 100%|██████████| 73/73 [00:21<00:00,  3.34it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:06<00:00,  3.47it/s]

                   all        166        224      0.814      0.823      0.842      0.471



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     29/100       1.8G      1.527      1.128      2.331         11        640: 100%|██████████| 73/73 [00:21<00:00,  3.37it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:06<00:00,  3.42it/s]

                   all        166        224      0.857      0.812      0.838      0.478



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     30/100      1.81G      1.507      1.105      2.276         10        640: 100%|██████████| 73/73 [00:21<00:00,  3.37it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:06<00:00,  3.43it/s]

                   all        166        224      0.872      0.794      0.816      0.463



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     31/100      1.83G       1.49      1.084      2.254         12        640: 100%|██████████| 73/73 [00:21<00:00,  3.34it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:05<00:00,  3.54it/s]

                   all        166        224      0.836      0.798      0.812      0.477



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     32/100      1.85G      1.546      1.161      2.317         11        640: 100%|██████████| 73/73 [00:21<00:00,  3.35it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:05<00:00,  4.16it/s]

                   all        166        224      0.876      0.817       0.86      0.483



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     33/100      1.87G      1.522       1.14      2.274         10        640: 100%|██████████| 73/73 [00:22<00:00,  3.21it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.35it/s]

                   all        166        224      0.815      0.848      0.828      0.483



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     34/100      1.87G       1.48      1.147      2.195         12        640: 100%|██████████| 73/73 [00:22<00:00,  3.22it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.69it/s]

                   all        166        224      0.857      0.804      0.839      0.493



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     35/100       1.9G      1.477       1.07      2.236         15        640: 100%|██████████| 73/73 [00:23<00:00,  3.07it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  5.16it/s]

                   all        166        224      0.824      0.795      0.819      0.468



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     36/100      1.92G      1.472      1.101      2.253          8        640: 100%|██████████| 73/73 [00:23<00:00,  3.11it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.86it/s]

                   all        166        224      0.886      0.796      0.843       0.49



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     37/100      1.94G      1.513      1.048      2.281          8        640: 100%|██████████| 73/73 [00:23<00:00,  3.10it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.94it/s]

                   all        166        224      0.892      0.811      0.851      0.489



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     38/100      1.96G      1.495      1.108      2.268         14        640: 100%|██████████| 73/73 [00:23<00:00,  3.09it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.81it/s]

                   all        166        224      0.833      0.804      0.836      0.489



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     39/100      1.97G      1.452       1.09      2.192         16        640: 100%|██████████| 73/73 [00:23<00:00,  3.10it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.62it/s]

                   all        166        224      0.809       0.83      0.849      0.494



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     40/100      1.99G      1.472      1.111      2.241         10        640: 100%|██████████| 73/73 [00:23<00:00,  3.05it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.69it/s]

                   all        166        224      0.864      0.821      0.848      0.497



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     41/100      2.01G      1.432       1.04      2.188         15        640: 100%|██████████| 73/73 [00:23<00:00,  3.07it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.94it/s]

                   all        166        224      0.865       0.83      0.865      0.511



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     42/100      2.02G        1.4      1.033      2.234         13        640: 100%|██████████| 73/73 [00:23<00:00,  3.10it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.79it/s]

                   all        166        224      0.869      0.835      0.864      0.513



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     43/100      2.04G      1.445      1.034      2.245         17        640: 100%|██████████| 73/73 [00:23<00:00,  3.08it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.78it/s]

                   all        166        224       0.85      0.839      0.841      0.506



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     44/100      2.06G      1.468      1.084      2.239         10        640: 100%|██████████| 73/73 [00:23<00:00,  3.09it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.98it/s]

                   all        166        224      0.842       0.83      0.839      0.505



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     45/100      2.08G      1.452      1.029      2.175          9        640: 100%|██████████| 73/73 [00:22<00:00,  3.21it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.25it/s]

                   all        166        224      0.863      0.816      0.846      0.498



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     46/100      2.09G      1.395      1.044       2.17         22        640: 100%|██████████| 73/73 [00:21<00:00,  3.36it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:05<00:00,  3.73it/s]

                   all        166        224      0.885       0.83      0.858      0.525



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     47/100      2.11G      1.432      1.035      2.201         11        640: 100%|██████████| 73/73 [00:21<00:00,  3.38it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:06<00:00,  3.49it/s]

                   all        166        224      0.874      0.835      0.875      0.527



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     48/100      2.13G      1.391      0.972      2.149         13        640: 100%|██████████| 73/73 [00:21<00:00,  3.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:05<00:00,  3.51it/s]

                   all        166        224      0.892      0.846      0.877      0.534



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     49/100      2.15G      1.358     0.9824      2.136          8        640: 100%|██████████| 73/73 [00:21<00:00,  3.39it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:05<00:00,  3.84it/s]

                   all        166        224      0.862      0.821      0.838      0.507



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     50/100      2.16G      1.355     0.9563      2.137         13        640: 100%|██████████| 73/73 [00:21<00:00,  3.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.32it/s]

                   all        166        224      0.857      0.862      0.861      0.533



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     51/100      2.18G       1.43     0.9813       2.22         11        640: 100%|██████████| 73/73 [00:21<00:00,  3.35it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  5.00it/s]

                   all        166        224      0.854      0.834      0.858      0.526



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     52/100       2.2G       1.35      1.007      2.156         13        640: 100%|██████████| 73/73 [00:22<00:00,  3.26it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.81it/s]

                   all        166        224      0.837      0.875      0.873      0.532



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     53/100      2.22G      1.336     0.9783        2.1         11        640: 100%|██████████| 73/73 [00:23<00:00,  3.13it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.98it/s]

                   all        166        224      0.843      0.841      0.854      0.519



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     54/100      2.24G      1.366     0.9544      2.129          8        640: 100%|██████████| 73/73 [00:22<00:00,  3.20it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.88it/s]

                   all        166        224      0.821      0.866      0.876      0.536



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     55/100      2.25G      1.382     0.9857      2.171         18        640: 100%|██████████| 73/73 [00:23<00:00,  3.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.87it/s]

                   all        166        224      0.865      0.839      0.865      0.509



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     56/100      2.27G      1.344     0.9601      2.128         12        640: 100%|██████████| 73/73 [00:22<00:00,  3.29it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.26it/s]

                   all        166        224      0.843      0.842       0.86      0.524



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     57/100      2.29G      1.372     0.9744      2.118          9        640: 100%|██████████| 73/73 [00:21<00:00,  3.38it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:05<00:00,  3.62it/s]

                   all        166        224      0.837      0.804      0.822      0.503



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     58/100      2.29G      1.367     0.9562      2.203         19        640: 100%|██████████| 73/73 [00:21<00:00,  3.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:05<00:00,  3.56it/s]

                   all        166        224       0.88      0.854       0.88      0.535



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     59/100      2.32G      1.353      0.946      2.142          8        640: 100%|██████████| 73/73 [00:21<00:00,  3.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:05<00:00,  3.78it/s]

                   all        166        224      0.912      0.857       0.88      0.534



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     60/100      2.34G      1.372     0.9836      2.151         14        640: 100%|██████████| 73/73 [00:21<00:00,  3.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.22it/s]

                   all        166        224      0.855      0.866      0.868      0.535



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     61/100      2.36G      1.323     0.9592        2.1          7        640: 100%|██████████| 73/73 [00:21<00:00,  3.36it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.79it/s]

                   all        166        224      0.858      0.837      0.873      0.531



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     62/100      2.38G      1.317     0.9343      2.103          8        640: 100%|██████████| 73/73 [00:22<00:00,  3.22it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.91it/s]

                   all        166        224      0.871      0.842       0.89      0.543



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     63/100       2.4G      1.333     0.9437      2.125         18        640: 100%|██████████| 73/73 [00:23<00:00,  3.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.82it/s]

                   all        166        224      0.888      0.854      0.877      0.531



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     64/100      2.41G      1.337     0.9491      2.117         12        640: 100%|██████████| 73/73 [00:23<00:00,  3.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  5.07it/s]

                   all        166        224      0.821      0.866      0.861      0.518



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     65/100      2.43G      1.339     0.9155      2.085          4        640: 100%|██████████| 73/73 [00:22<00:00,  3.19it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.98it/s]

                   all        166        224      0.828      0.857      0.879      0.544



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     66/100      2.44G      1.276     0.9039      2.061          8        640: 100%|██████████| 73/73 [00:22<00:00,  3.22it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.63it/s]

                   all        166        224       0.87      0.838      0.873      0.527



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     67/100      2.47G      1.302      0.921      2.128         11        640: 100%|██████████| 73/73 [00:21<00:00,  3.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:05<00:00,  3.83it/s]

                   all        166        224      0.878      0.838      0.872      0.535



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     68/100      2.48G      1.294     0.8966      2.097          8        640: 100%|██████████| 73/73 [00:21<00:00,  3.47it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:06<00:00,  3.46it/s]

                   all        166        224      0.903      0.848      0.876      0.537



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     69/100       2.5G      1.322     0.9141      2.079         14        640: 100%|██████████| 73/73 [00:21<00:00,  3.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:05<00:00,  3.70it/s]

                   all        166        224      0.913      0.866      0.887      0.542



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     70/100      2.52G      1.324     0.9362      2.124         12        640: 100%|██████████| 73/73 [00:21<00:00,  3.39it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.62it/s]

                   all        166        224      0.861      0.871      0.875      0.544



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     71/100      2.54G      1.249     0.8982      2.075         10        640: 100%|██████████| 73/73 [00:21<00:00,  3.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.83it/s]

                   all        166        224       0.88      0.888      0.874      0.544



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     72/100      2.55G      1.291     0.9287      2.077         16        640: 100%|██████████| 73/73 [00:22<00:00,  3.24it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  5.16it/s]

                   all        166        224      0.851      0.888      0.885      0.554



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     73/100      2.57G      1.253     0.8978      2.013         13        640: 100%|██████████| 73/73 [00:23<00:00,  3.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  5.13it/s]

                   all        166        224      0.891      0.839      0.896       0.55



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     74/100      2.58G      1.277     0.8668      2.074         17        640: 100%|██████████| 73/73 [00:23<00:00,  3.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.98it/s]

                   all        166        224      0.881      0.829      0.885      0.545



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     75/100      2.61G      1.264     0.8849      2.009         13        640: 100%|██████████| 73/73 [00:23<00:00,  3.11it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.83it/s]

                   all        166        224      0.895      0.866      0.883      0.534



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     76/100      2.62G      1.213     0.8649      2.001          7        640: 100%|██████████| 73/73 [00:23<00:00,  3.05it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.76it/s]

                   all        166        224      0.886      0.869      0.883      0.545



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     77/100      2.64G       1.25     0.8943      1.989         18        640: 100%|██████████| 73/73 [00:23<00:00,  3.07it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.77it/s]

                   all        166        224      0.885      0.853      0.864       0.53



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     78/100      2.66G      1.243     0.8708      1.994         16        640: 100%|██████████| 73/73 [00:23<00:00,  3.12it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.64it/s]

                   all        166        224      0.866      0.871      0.866       0.54



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     79/100      2.67G      1.237     0.8496      1.994         17        640: 100%|██████████| 73/73 [00:23<00:00,  3.12it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.66it/s]

                   all        166        224      0.846      0.884      0.879      0.559



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     80/100      2.69G      1.197     0.8341      1.992          2        640: 100%|██████████| 73/73 [00:22<00:00,  3.20it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:05<00:00,  3.95it/s]

                   all        166        224      0.857      0.888      0.903      0.573



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     81/100      2.71G      1.187     0.8211      1.938         14        640: 100%|██████████| 73/73 [00:22<00:00,  3.26it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:05<00:00,  3.59it/s]

                   all        166        224      0.878      0.875      0.894      0.568



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     82/100      2.72G       1.27     0.8594      2.061          9        640: 100%|██████████| 73/73 [00:22<00:00,  3.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:06<00:00,  3.40it/s]

                   all        166        224      0.903      0.877      0.908      0.569



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     83/100      2.75G      1.205     0.8394      1.931         10        640: 100%|██████████| 73/73 [00:21<00:00,  3.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:06<00:00,  3.33it/s]

                   all        166        224      0.861      0.884       0.89      0.565



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     84/100      2.77G      1.157     0.8253      1.923         21        640: 100%|██████████| 73/73 [00:21<00:00,  3.38it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:06<00:00,  3.28it/s]

                   all        166        224      0.873      0.884      0.892       0.57



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     85/100      2.78G       1.24     0.8564      2.021          7        640: 100%|██████████| 73/73 [00:22<00:00,  3.28it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:06<00:00,  3.50it/s]

                   all        166        224      0.876      0.893      0.899      0.567



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     86/100       2.8G      1.195     0.8457      2.009         17        640: 100%|██████████| 73/73 [00:22<00:00,  3.29it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:05<00:00,  3.62it/s]

                   all        166        224      0.864      0.875      0.895       0.56



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     87/100      2.82G      1.174     0.8201      1.988         13        640: 100%|██████████| 73/73 [00:22<00:00,  3.29it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:05<00:00,  3.51it/s]

                   all        166        224      0.874      0.888      0.897      0.575



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     88/100      2.84G      1.216     0.8425      1.967         12        640: 100%|██████████| 73/73 [00:21<00:00,  3.35it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:05<00:00,  3.69it/s]

                   all        166        224      0.891      0.875      0.885      0.572



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     89/100      2.85G      1.154     0.8292      1.925          4        640: 100%|██████████| 73/73 [00:22<00:00,  3.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:05<00:00,  3.82it/s]

                   all        166        224      0.869      0.879      0.877      0.563



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     90/100      2.86G      1.151      0.796      1.934         10        640: 100%|██████████| 73/73 [00:22<00:00,  3.26it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.27it/s]

                   all        166        224      0.882      0.867      0.879      0.559


Closing dataloader mosaic
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, num_output_channels=3, method='weighted_average'), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     91/100      2.89G      1.032     0.7253      1.875          7        640: 100%|██████████| 73/73 [00:23<00:00,  3.11it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:05<00:00,  3.66it/s]

                   all        166        224      0.863      0.869      0.875      0.556



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     92/100       2.9G      1.007     0.7083      1.854          7        640: 100%|██████████| 73/73 [00:21<00:00,  3.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.34it/s]

                   all        166        224      0.879      0.888      0.895      0.566



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     93/100      2.92G      1.007      0.679      1.861          6        640: 100%|██████████| 73/73 [00:22<00:00,  3.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.83it/s]

                   all        166        224      0.878      0.879      0.888      0.566



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     94/100      2.94G     0.9957     0.6593      1.833          7        640: 100%|██████████| 73/73 [00:22<00:00,  3.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  5.23it/s]

                   all        166        224      0.891      0.874      0.889      0.567



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     95/100      2.96G     0.9878     0.6714      1.834          9        640: 100%|██████████| 73/73 [00:22<00:00,  3.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:03<00:00,  5.26it/s]

                   all        166        224      0.887      0.866      0.885      0.563



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     96/100      2.97G      1.007     0.6689      1.859          6        640: 100%|██████████| 73/73 [00:22<00:00,  3.28it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.21it/s]

                   all        166        224      0.889      0.862      0.886       0.56



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     97/100      2.99G     0.9853      0.657       1.84          7        640: 100%|██████████| 73/73 [00:24<00:00,  3.01it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:04<00:00,  4.26it/s]

                   all        166        224      0.896      0.866      0.888      0.566



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     98/100         3G     0.9727     0.6441      1.798          9        640: 100%|██████████| 73/73 [00:21<00:00,  3.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:05<00:00,  3.57it/s]

                   all        166        224      0.889      0.857      0.892      0.574



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     99/100      3.03G     0.9692     0.6475      1.827          5        640: 100%|██████████| 73/73 [00:21<00:00,  3.38it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:06<00:00,  3.20it/s]

                   all        166        224      0.885      0.862      0.892      0.574



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    100/100      3.04G     0.9625     0.6364      1.793          5        640: 100%|██████████| 73/73 [00:21<00:00,  3.47it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:06<00:00,  3.48it/s]

                   all        166        224      0.892      0.866      0.893      0.575



100 epochs completed in 0.786 hours.
Optimizer stripped from deteccion_balsas/yolov8n_obb_run1/weights/last.pt, 6.5MB
Optimizer stripped from deteccion_balsas/yolov8n_obb_run1/weights/best.pt, 6.5MB

Validating deteccion_balsas/yolov8n_obb_run1/weights/best.pt...
Ultralytics 8.3.102 🚀 Python-3.11.11 torch-2.6.0+cu124 CUDA:0 (Tesla T4, 15095MiB)
YOLOv8n-obb summary (fused): 81 layers, 3,077,414 parameters, 0 gradients, 8.3 GFLOPs


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:06<00:00,  3.34it/s]


                   all        166        224      0.883      0.879      0.897      0.574
Speed: 0.4ms preprocess, 6.5ms inference, 0.0ms loss, 9.8ms postprocess per image
Results saved to deteccion_balsas/yolov8n_obb_run1
CPU times: user 22min 24s, sys: 1min 19s, total: 23min 44s
Wall time: 53min 43s


## Evaluación del modelo con el Conjunto de Validación (val)  

El conjunto de validación se utiliza típicamente durante el entrenamiento para monitorear el rendimiento del modelo. Puedes evaluar el modelo en este conjunto usando el método model.val().

In [13]:
# Evaluar el modelo en el conjunto de validación
results_val = model.val(plots=True)   # Evalúa en el conjunto de test configurado en data.yaml

# Imprimir métricas clave
print("Resultados en el conjunto de validación:")
print(f"mAP@0.5: {results_val.box.map50}")  # mAP con IoU=0.5
print(f"mAP@0.5:0.95: {results_val.box.map}")  # mAP con IoU entre 0.5 y 0.95

Ultralytics 8.3.102 🚀 Python-3.11.11 torch-2.6.0+cu124 CUDA:0 (Tesla T4, 15095MiB)
YOLOv8n-obb summary (fused): 81 layers, 3,077,414 parameters, 0 gradients, 8.3 GFLOPs


val: Scanning /content/drive/My Drive/TFM_BALSAS/dataset/labels/val.cache... 166 images, 0 backgrounds, 0 corrupt: 100%|██████████| 166/166 [00:00<?, ?it/s]

val: WARNING ⚠️ /content/drive/My Drive/TFM_BALSAS/dataset/images/val/259.tif: 1 duplicate labels removed
val: WARNING ⚠️ /content/drive/My Drive/TFM_BALSAS/dataset/images/val/594.tif: 1 duplicate labels removed



                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:06<00:00,  3.12it/s]


                   all        166        224      0.883      0.879      0.897      0.574
Speed: 2.7ms preprocess, 8.9ms inference, 0.0ms loss, 7.1ms postprocess per image
Results saved to deteccion_balsas/yolov8n_obb_run1
Ultralytics 8.3.102 🚀 Python-3.11.11 torch-2.6.0+cu124 CUDA:0 (Tesla T4, 15095MiB)


val: Scanning /content/drive/My Drive/TFM_BALSAS/dataset/labels/val.cache... 166 images, 0 backgrounds, 0 corrupt: 100%|██████████| 166/166 [00:00<?, ?it/s]

val: WARNING ⚠️ /content/drive/My Drive/TFM_BALSAS/dataset/images/val/259.tif: 1 duplicate labels removed
val: WARNING ⚠️ /content/drive/My Drive/TFM_BALSAS/dataset/images/val/594.tif: 1 duplicate labels removed



                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 21/21 [00:06<00:00,  3.11it/s]


                   all        166        224      0.883      0.879      0.897      0.574
Speed: 3.8ms preprocess, 7.8ms inference, 0.0ms loss, 8.6ms postprocess per image
Results saved to deteccion_balsas/yolov8n_obb_run1
Resultados en el conjunto de validación:
mAP@0.5: 0.8972523797677318
mAP@0.5:0.95: 0.5738130941727748


## Valoración de los resultados

Los resultadosp roporcionados por la evaluación muestran un desempeño muy bueno del modelo **YOLOv8n-obb** en el conjunto de validación. REsumimos a continuación los puntos más destacados:

### Métricas clave
Las métricas principales son:

- **mAP@0.5 (mean Average Precision a IoU=0.5):**  
  - Valor: **0.897** (~89.7%)  
  - Significado: Indica la precisión promedio del modelo cuando se considera que una predicción es correcta si su IoU (Intersection over Union) con la caja real es al menos del 50%.  
  - Evaluación: Este valor es excelente. Un mAP@0.5 superior al 85% sugiere que el modelo está detectando objetos con alta precisión.

- **mAP@0.5:0.95 (mean Average Precision a IoU entre 0.5 y 0.95):**  
  - Valor: **0.574** (~57.4%)  
  - Significado: Es una métrica más estricta que evalúa la precisión promedio en diferentes umbrales de IoU (desde 0.5 hasta 0.95).  
  - Evaluación: Este valor también es sólido. Un mAP@0.5:0.95 superior al 50% indica que el modelo no solo detecta bien las cajas, sino que también localiza los objetos con precisión.

- **Precisión (P):**  
  - Valor: **0.883** (~88.3%)  
  - Significado: Proporción de predicciones correctas entre todas las predicciones realizadas.  
  - Evaluación: Una precisión del 88.3% es muy buena, lo que indica que el modelo tiene pocas falsas alarmas.

- **Recall (R):**  
  - Valor: **0.879** (~87.9%)  
  - Significado: Proporción de objetos reales detectados correctamente.  
  - Evaluación: Un recall del 87.9% es excelente, lo que sugiere que el modelo detecta la mayoría de los objetos presentes en las imágenes.

### Conclusión
- **Desempeño:**  
  El modelo ha alcanzado métricas muy buenas en el conjunto de validación, con un mAP@0.5 del **89.7%** y un mAP@0.5:0.95 del **57.4%**. Estos valores indican que el modelo es capaz de detectar y localizar objetos con alta precisión.

- **Velocidad:**  
  La inferencia es rápida (~20 ms por imagen).

En resumen, los resultados obtenidos son **muy buenos** y demuestran que el modelo YOLOv8n-obb está funcionando muy bien para la tarea de detección de balsas-obb. Con una limpieza adicional de los datos y posibles ajustes en el entrenamiento se podrías alcanzar métricas mejores. 